app/store/migrations/052.test.ts

```diff
@@ -0,0 +1,199 @@
+            AccountsController: { internalAccounts: null },
+          },
+        },
+      }),
+      scenario: 'AccountsController internalAccounts state is invalid',
+      expectedError:
+        'FATAL ERROR: Migration 52: Invalid AccountsController state',
+    },
+    {
+      state: merge({}, initialRootState, {
+        engine: {
+          backgroundState: {
+            AccountsController: {
+              internalAccounts: {
+                accounts: null,
+              },
+            },
+          },
+        },
+      }),
+      scenario: 'AccountsController internalAccounts accounts state is invalid',
+      expectedError:
+        'FATAL ERROR: Migration 52: Invalid AccountsController state',
+    },
+    {
+      state: merge({}, initialRootState, {
+        engine: {
+          backgroundState: {
+            AccountsController: {
+              internalAccounts: {
+                accounts: {},
+                selectedAccount: null,
+              },
+            },
+          },
+        },
+      }),
+      scenario:
+        'AccountsController internalAccounts selectedAccount is not a string',
+      expectedError:
+        'FATAL ERROR: Migration 52: Invalid AccountsController state',
+    },
+  ];
+
+  for (const { scenario, state, expectedError } of invalidStates) {
+    it(`captures exception if ${scenario}`, () => {
+      const newState = migration(state);
+
+      expect(newState).toStrictEqual(state);
+      expect(mockedCaptureException).toHaveBeenCalledWith(expect.any(Error));
+      expect(mockedCaptureException.mock.calls[0][0].message).toBe(
+        expectedError,
+      );
+    });
+  }
+
+  it('does not change the selectedAccount if it is valid', () => {
+    const newState = migration(oldState) as typeof oldState;
+
+    expect(
+      newState.engine.backgroundState.AccountsController.internalAccounts
+        .selectedAccount,
+    ).toEqual(expectedUuid);
+  });
+
+  it('updates the selectedAccount if it is invalid', () => {
+    const invalidState = merge({}, oldState, {
+      engine: {
+        backgroundState: {
+          AccountsController: {
+            internalAccounts: {
+              selectedAccount: 'invalid-uuid',
+            },
+          },
+        },
+      },
+    });
+
+    const newState = migration(invalidState) as typeof invalidState;
+
+    expect(
+      newState.engine.backgroundState.AccountsController.internalAccounts
+        .selectedAccount,
+    ).toEqual(expectedUuid);
+  });
+
+  it('does not change the state if there are no accounts', () => {
+    const emptyAccountsState = merge({}, oldState, {
+      engine: {
+        backgroundState: {
+          AccountsController: {
+            internalAccounts: {
+              accounts: {},
+              selectedAccount: 'some-uuid',
+            },
+          },
+        },
+      },
+    });
+
+    const newState: Pick<RootState, 'engine'> = migration(
+      emptyAccountsState,
+    ) as Pick<RootState, 'engine'>;
```

Comment:
```suggestion
    ) as typeof emptyAccountsState;
```
Does this make sense instead of typing with the current engine state? My intention with this comment is that we are more future proof on migrations, and with updating on the future, do not break this types, since the engine structure could change on the future

Reply:
gotcha! addressed here [4289304](https://github.com/MetaMask/metamask-mobile/pull/10821/commits/4289304745d4fa7a7e1e302eee1cc51a99f829b6)